pkg/config/mapper.go

```diff
@@ -0,0 +1,100 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/logandavies181/arnlike"
+)
+
+// Validate returns an error if the RoleMapping is not valid after being unmarshaled
+func (m *RoleMapping) Validate() error {
+	if m == nil {
+		return fmt.Errorf("RoleMapping is nil")
+	}
+
+	if m.RoleARN == "" && m.RoleARNLike == "" {
+		return fmt.Errorf("One of rolearn or rolearnLike must be supplied")
+	} else if m.RoleARN != "" && m.RoleARNLike != "" {
+		return fmt.Errorf("Only one of rolearn or rolearnLike can be supplied")
+	}
+
+	if m.RoleARNLike != "" {
+		ok, err := arnlike.ArnLike(m.RoleARNLike, "arn:*:iam:*:*:role/*")
+		if err != nil {
+			return err
+		} else if !ok {
+			return fmt.Errorf("RoleARNLike '%s' did not match an ARN for an IAM Role", m.RoleARNLike)
+		}
+	}
+
+	return nil
+}
+
+// Matches returns true if the supplied ARN or ARN-like string matches
+// this RoleMapping
+func (m *RoleMapping) Matches(subject string) bool {
+	if m.RoleARN != "" {
+		return strings.ToLower(m.RoleARN) == strings.ToLower(subject)
```

Comment:
@jaypipes I'm not 100% sure why previously RoleARNs have been lowercased before being put in maps, but anyway I've moved all the lowercasing to this comparison so the unit tests still pass

Reply:
Yup, no worries. ++